zerver/lib/soft_deactivation.py

```diff
@@ -369,12 +369,16 @@ def do_catch_up_soft_deactivated_users(users: Iterable[UserProfile]) -> list[Use
     failures = []
     for user_profile in users:
         if user_profile.long_term_idle:
-            try:
-                add_missing_messages(user_profile)
-                users_caught_up.append(user_profile)
-            except Exception:  # nocoverage
-                capture_exception()  # nocoverage
-                failures.append(user_profile)  # nocoverage
+            with sentry_sdk.isolation_scope() as scope:
+                scope.set_user({"id": str(user_profile.id)})
+                try:
+                    add_missing_messages(user_profile)
+                    users_caught_up.append(user_profile)
+                except Exception:  # nocoverage
+                    logger.exception(
+                        "Failed to catch up %d@%s", user_profile.id, user_profile.realm.string_id
+                    )  # nocoverage
+                    failures.append(user_profile)  # nocoverage
```

Comment:
This is not new, but I think only the `nocoverage` comment on `except Exception` is operative.

Reply:
Tacked on a commit.